core/lib/src/request/from_request.rs

```diff
@@ -312,3 +314,58 @@ impl<'a, 'r, T: FromRequest<'a, 'r>> FromRequest<'a, 'r> for Option<T> {
     }
 }
 
+#[cfg(feature = "tls")]
+impl <'a, 'r> FromRequest<'a, 'r> for MutualTlsUser {
+    type Error = ();
+
+    fn from_request(request: &'a Request<'r>) -> Outcome<Self, Self::Error> {
+        // Get peer's IP address
+        let ip_addr = match request.client_ip() {
```

Comment:
I think you can do something like:

```rust
let ip_addr = request.client_ip().or_forward(())?;
```

Use this here and everywhere else in this implementation.